yktrace/src/hwt/mapper.rs

```diff
@@ -51,20 +51,21 @@ impl BlockMap {
         while crsr.position() < sec_size {
             let f_off = crsr.read_u64::<NativeEndian>().unwrap();
             let n_blks = leb128::read::unsigned(&mut crsr).unwrap();
-            for bb in 0..n_blks {
+            for _ in 0..n_blks {
                 let b_off = leb128::read::unsigned(&mut crsr).unwrap();
                 // Skip the block size. We still have to parse the field, as it's variable-size.
                 let b_sz = leb128::read::unsigned(&mut crsr).unwrap();
                 // Skip over block meta-data.
                 crsr.seek(SeekFrom::Current(1)).unwrap();
+                let b_idx = leb128::read::unsigned(&mut crsr).unwrap();
```

Comment:
So do we definitely have a strategy for dealing with the case where a `BasicBlock` lowers to many `MachineBasicBlock`s?

Can you think of any scenario where it could be ambiguous?

For example, if a trace reports the blocks `<1, 2, 1, 1, 4>`, do we always know how to interpret `1, 1`? This could either be a block with a call, or a block which lowered to 2 MBBs.

Reply:
> For example, if a trace reports the blocks `<1, 2, 1, 1, 4>`, do we always know how to interpret `1, 1`? This could either be a block with a call, or a block which lowered to 2 MBBs.

Good point. I think we might be able to encode this into the address map, i.e. when we a machine block maps to an IR block we've already seen, then don't emit the index. This way we know when we see an MBB without an index, that it comes from a split BB. Though I'm not sure if there are other surprises with split up BBs but we won't know until we see some in action.

Reply:
OK, sounds like we will have to cross that bridge if and when we come to it.

Reply:
Can we not test for this at the moment because we don't support function calls?

Reply:
Yeah, we can't do function calls atm.

Reply:
OK.